File: pgpainless-sop/src/main/java/org/pgpainless/sop/ListProfilesImpl.java
// SPDX-FileCopyrightText: 2023 Paul Schaub <vanitasvitae@fsfe.org>
//
// SPDX-License-Identifier: Apache-2.0

package org.pgpainless.sop;

import java.util.List;

import sop.Profile;
import sop.exception.SOPGPException;
import sop.operation.ListProfiles;

import javax.annotation.Nonnull;

/**
 * Implementation of the <pre>list-profiles</pre> operation using PGPainless.
 *
 */
public class ListProfilesImpl implements ListProfiles {

    @Override
    @Nonnull
    public List<Profile> subcommand(@Nonnull String command) {

        switch (command) {
            case "generate-key":
                return GenerateKeyImpl.SUPPORTED_PROFILES;

            case "encrypt":
                return EncryptImpl.SUPPORTED_PROFILES;

            default:
                throw new SOPGPException.UnsupportedProfile(command);
        }
    }
}

File: pgpainless-sop/src/main/kotlin/org/pgpainless/sop/ListProfilesImpl.kt
// SPDX-FileCopyrightText: 2024 Paul Schaub <vanitasvitae@fsfe.org>
//
// SPDX-License-Identifier: Apache-2.0

package org.pgpainless.sop

import sop.Profile
import sop.exception.SOPGPException
import sop.operation.ListProfiles

/** Implementation of the `list-profiles` operation using PGPainless. */
class ListProfilesImpl : ListProfiles {

    override fun subcommand(command: String): List<Profile> =
        when (command) {
            "generate-key" -> GenerateKeyImpl.SUPPORTED_PROFILES
            "encrypt" -> EncryptImpl.SUPPORTED_PROFILES
            else -> throw SOPGPException.UnsupportedProfile(command)
        }
}
